rcl/test/rcl/test_subscription.cpp

```diff
@@ -885,6 +886,431 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_bad_return_l
+      rcl_get_zero_initialized_subscription_content_filter_options();
+
+    ret = rcl_subscription_get_content_filter(
+      &subscription, &content_filter_options);
+    if (is_cft_support) {
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+
+      rmw_subscription_content_filter_options_t * options =
+        &content_filter_options.rmw_subscription_content_filter_options;
+      ASSERT_STREQ(filter_expression2, options->filter_expression);
+      ASSERT_EQ(expression_parameters2_count, options->expression_parameters.size);
+      for (size_t i = 0; i < expression_parameters2_count; ++i) {
+        EXPECT_STREQ(
+          options->expression_parameters.data[i],
+          expression_parameters2[i]);
+      }
+      EXPECT_EQ(
+        RCL_RET_OK,
+        rcl_subscription_content_filter_options_fini(
+          &content_filter_options)
+      );
+    } else {
+      ASSERT_EQ(RCL_RET_UNSUPPORTED, ret);
+    }
+  }
+
+  // reset filter
+  {
+    rcl_subscription_content_filter_options_t options =
+      rcl_get_zero_initialized_subscription_content_filter_options();
+
+    EXPECT_EQ(
+      RCL_RET_OK,
+      rcl_subscription_content_filter_options_init(
+        "", 0, nullptr,
+        &options)
+    );
+
+    ret = rcl_subscription_set_content_filter(
+      &subscription, &options);
+    if (is_cft_support) {
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+      // waiting to allow for filter propagation
+      std::this_thread::sleep_for(std::chrono::seconds(1));
+      ASSERT_TRUE(wait_for_established_subscription(&publisher, 30, 100));
+      ASSERT_FALSE(rcl_subscription_is_cft_enabled(&subscription));
+    } else {
+      ASSERT_EQ(RCL_RET_UNSUPPORTED, ret);
+    }
+
+    EXPECT_EQ(
+      RCL_RET_OK,
+      rcl_subscription_content_filter_options_fini(
+        &options)
+    );
+  }
+
+  // publish with a non-filtered data again
+  {
+    test_msgs__msg__Strings msg;
+    test_msgs__msg__Strings__init(&msg);
+    ASSERT_TRUE(rosidl_runtime_c__String__assign(&msg.string_value, test_string));
+    ret = rcl_publish(&publisher, &msg, nullptr);
+    test_msgs__msg__Strings__fini(&msg);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  }
+
+  ASSERT_TRUE(wait_for_subscription_to_be_ready(&subscription, context_ptr, 30, 100));
+
+  {
+    test_msgs__msg__Strings msg;
+    test_msgs__msg__Strings__init(&msg);
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      test_msgs__msg__Strings__fini(&msg);
+    });
+    ret = rcl_take(&subscription, &msg, nullptr, nullptr);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ASSERT_EQ(
+      std::string(test_string),
+      std::string(msg.string_value.data, msg.string_value.size));
+  }
+}
+
+/* A subscription without a content filtered topic setting at beginning.
+ */
+TEST_F(
+  CLASSNAME(
+    TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription_not_content_filtered_at_begin) {
```

Comment:
```suggestion
    TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription_not_initialized_with_content_filtering) {
```

Reply:
Thanks, updated in 206facde4e4621d2af045e3ff176758f81ad6fda